core-rust/state-manager/src/state_manager.rs

```diff
@@ -890,25 +891,53 @@ where
     }
 }
 
-struct StateTracker {
+#[derive(Debug, Clone)]
+struct StateSummary {
+    state_version: u64,
     accumulator_hash: AccumulatorHash,
+}
+
+impl StateSummary {
+    pub fn from(committed_ids: Option<CommittedTransactionIdentifiers>) -> Self {
+        let ids = committed_ids.unwrap_or_else(CommittedTransactionIdentifiers::pre_genesis);
+        Self {
+            state_version: ids.state_version,
+            accumulator_hash: ids.accumulator_hash,
+        }
+    }
+
+    pub fn pre_genesis() -> Self {
+        Self::from(None)
+    }
+
+    pub fn to_committed_identifiers(&self) -> CommittedTransactionIdentifiers {
+        CommittedTransactionIdentifiers {
+            state_version: self.state_version,
+            accumulator_hash: self.accumulator_hash,
+        }
+    }
+}
+
+struct StateTracker {
+    summary: StateSummary,
```

Comment:
why don't we reuse `CommittedTransactionIdentifiers`? f.e. `latest_transaction_identifiers: CommittedTransactionIdentifiers`?

Reply:
ahh, I had the need for "state summary" (in the logic) and then I didn't notice it is 100% identical to "transaction identifiers" - you are right, no point copying 👍 

(I unified the name of `latest_state_hash()` too)